service_manager/android/src/main/java/com/baseflow/service_manager/EventStreamHandlerImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.baseflow.service_manager;
+
+import android.bluetooth.BluetoothAdapter;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+
+import io.flutter.plugin.common.EventChannel;
+
+public class EventStreamHandlerImpl implements EventChannel.StreamHandler {
+
+    private final ServiceManager serviceManager;
+    private final Context applicationContext;
+
+    private BroadcastReceiver stateReceiver;
+    private EventChannel.EventSink eventSink;
+
+    public EventStreamHandlerImpl(ServiceManager serviceManager, Context applicationContext) {
+
+        this.serviceManager = serviceManager;
+        this.applicationContext = applicationContext;
+
+        stateReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                if (eventSink == null) {
+                    return;
+                }
+
+                final int state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, -1);
+                switch (state) {
```

Comment:
I would add turning_on and turning_off as well. Maybe even STATE_CONNECTED and so on. All the states from here:
https://developer.android.com/reference/android/bluetooth/BluetoothAdapter#STATE_CONNECTED